### PR TITLE
Add onclick events

### PR DESCRIPTION
**Pull request by RokoVarano**
Onclick events have been added.

I have a question bout the requirements: on clicking numbers, the number displayed is not updated. Only when clicking the operations the total updates. Is this the intended behaviour? or should the number update to match the button that has just been clicked?
Could you describe the behaviour of the display?
Thanks in advance

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,17 +1,30 @@
 import React, { Component } from 'react';
 import Button from './Button';
+import calculate from '../logic/calculate';
```
Comment (arslanbisharat) [inline src/components/Calculator.js:4]: - [x] While I can see you have imported calculate.js from the logic directory but I cannot see operate.js being imported here. Please import it as we need it here to run them.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 
   render() {
+    const { total } = this.state;
     return (
       <section className="mainframe">
-        <p className="result">0</p>
-        {this.buttons.map((text) => <Button content={text} key={`btn-${text}`} />)}
+        <p className="result">{ total }</p>
+        {this.buttons.map((text) => <Button content={text} key={`btn-${text}`} click={() => this.onClickButton(this.state, text)} />)}
       </section>
```
Comment (arslanbisharat) [inline src/components/Calculator.js]: - [ ] No button(click) is working for me on the live version of the project. Please make sure the logic works fine for this project.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Calculator.js (hunk 1 above)
Comment (RokoVarano) [reply src/components/Calculator.js:4]: Could you explain please? operate.js is used by calculate.js. What is it that I need to do with operate.js?

File: src/components/Calculator.js (hunk 2 above)
Comment (RokoVarano) [reply src/components/Calculator.js]: As I asked in the original post, could you explain the expected behaviour? Could you explain what is it that doesn't work?

Comment (arslanbisharat) [reply src/components/Calculator.js]: Hi there , 
It should work just like a calculator but for your project I cannot use button. It's mean handle click has a problem.

File: src/components/Calculator.js (hunk 1 above)
Comment (arslanbisharat) [reply src/components/Calculator.js:4]: I consulted with TSE leader and please ignore this requirement as it's already been imported. 
Thank you for the catch

**Comment by RokoVarano** [comment]
As many other Microverse students have mentioned in slack, calculate.js is broken and the right functionality cannot be achieved without modifying it, which goes against the requirements. Please consider this and ask other TSE's before requiring changes.

**Comment by arslanbisharat** [comment]
> As many other Microverse students have mentioned in slack, calculate.js is broken and the right functionality cannot be achieved without modifying it, which goes against the requirements. Please consider this and ask other TSE's before requiring changes.

Hi there, 
Thank you so much for reaching out. I would be happy to assist you with this matter. 
I have Reviewed many projects and buttons are functional there and calculate.js is not broken.

**Comment by RokoVarano** [comment]
> > As many other Microverse students have mentioned in slack, calculate.js is broken and the right functionality cannot be achieved without modifying it, which goes against the requirements. Please consider this and ask other TSE's before requiring changes.
> 
> Hi there,
> Thank you so much for reaching out. I would be happy to assist you with this matter.
> I have Reviewed many projects and buttons are functional there and calculate.js is not broken.

Have you tried clicking an operation button twice in a row in those projects?
if so, with the current calculate.js, there is an invalid number error because it is trying to get a null second number, which breaks it.
Solving this would be a lot of logic on the front end.

**Comment by arslanbisharat** [comment]
> > > As many other Microverse students have mentioned in slack, calculate.js is broken and the right functionality cannot be achieved without modifying it, which goes against the requirements. Please consider this and ask other TSE's before requiring changes.
> > 
> > 
> > Hi there,
> > Thank you so much for reaching out. I would be happy to assist you with this matter.
> > I have Reviewed many projects and buttons are functional there and calculate.js is not broken.
> 
> Have you tried clicking an operation button twice in a row in those projects?
> if so, with the current calculate.js, there is an invalid number error because it is trying to get a null second number, which breaks it.
> Solving this would be a lot of logic on the front end.

If you feel your project is not reviewed properly please ask for second opinion using the form link above.

File: README.md [hunk 3]
```diff
@@ -30,11 +34,14 @@ To get a local copy up and running follow these simple example steps.
...
-- coming soon
+- click on a number button, then on an operation to get the result displayed on the grey bar.
+- click on A/C to clear all operations.
 ### Run tests
 - No tests yet
 ## I learned:
 - how to setup a react project using create-react-app
+- how to use states and props to pass information between components
+- how to handle onClick events
 ## Authors
```
Comment (MahmoudBakr23) [inline README.md:45]: ### **OPTIONAL**

_For a completely professional README file and repo, please consider this:_
- [x] Kindly fix the MIT and issues page broken links
_This should have been brought up earlier, I'm sorry about that!_ 😄 🙏

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @RokoVarano 😃, 

Well done making it all the way up to this point 💯 Everything works fine to me 🥳 Good luck with the next 🙏 
Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by acushlakoncept — COMMENTED** [review]
Hello @RokoVarano 👋,

_This is an additional review as a result of the second opinion request._

From what I see here, I think it's a **problem of clear communication**. 
Reviewing from the commit head - `8e0bcc9` I experienced what @arslanbisharat talked about - the `handleClick` function having a bit of a bug I think, except your logic works fine to me.
When digits are clicked the display does not appear immediately until you click an operation sign, however, I think you managed to fix the issue on your next commit `9f20fcc`

I am glad to see your project has been approved already 😄 

Keep rocking 🚀!

**Pull request merged**